### PR TITLE
CI temporarily pin numpy

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Looks like numba is failing with the latest numpy release https://github.com/numba/numba/issues/8615

Let's temporarily pin numpy to get CI green, and then lift the pin once it's solved in numba?

### CODE REVIEW

**Comment by phofl** [comment]
I guess you have to pin in the asv config as well?

File: asv_bench/asv.conf.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     // pip (with all the conda available packages installed first,
     // followed by the pip installed packages).
     "matrix": {
-        "numpy": [],
+        "numpy": ["1.23.5"],  // https://github.com/pandas-dev/pandas/pull/50356
```
Comment (MarcoGorelli) [inline asv_bench/asv.conf.json:44]: asv doesn't allow pinning with `<`

Comment (phofl) [reply asv_bench/asv.conf.json:44]: Yeah I'd just go with the newest version as well.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @MarcoGorelli

**Comment by jorisvandenbossche** [comment]
Could we remove numba in some of the envs instead? Because not testing with the latest numpy anywhere doesn't seem like a good situation? 
(I don't know the timeline of a fix fox this in numba, but the issue is already open a while)

**Comment by MarcoGorelli** [comment]
the 311-dev job tests with the latest (nightly) numpy, and that one doesn't have numba

perhaps let's wait a few days to see if there's any progress, and if not just remove numba from CI 'til they resolve it?

**Pull request merged**